clis/tcli/src/bin/dtcli/daemon.rs

```diff
@@ -0,0 +1,119 @@
+//! Code for general daemon handling utilities and abstracting away related dependencies.
+
+use std::path::PathBuf;
+
+use anyhow::{self, Result};
+use sysinfo::{ProcessRefreshKind, RefreshKind, System};
+
+use crate::DTCLI_WD_PATH;
+
+/// Enum for figuring out the current process after the daemon is forked away
+/// from the API by the daemonization.
+pub enum ProcessType {
+    /// Getting this value after the daemonization means that the current process is the API.
+    Api,
+    /// Getting this value after the daemonization means that the current process is the Daemon.
+    Daemon,
+}
+
+/// Struct for packaging together the functions for managing the daemon.
+/// Currently does not need any members, since most of the data is stored
+/// in files anyway.
+pub struct Daemon;
+
+impl Daemon {
+    /// Checks whether the daemon is currently running.
+    /// Won't work if pid file had been tampered with or it's location changed.
+    ///
+    /// # Returns
+    ///
+    /// Result containing a bool. True means that the daemon is running.
+    pub fn is_running() -> Result<bool> {
```

Comment:
But doing it this way allows writing cleaner app code like `if is_running()? {/*do something*/} instead of matching everywhere to see if there's an issue with the system. Also a lot of the times the expected state is false so I wouldn't want to treat that as an error in code.

Reply:
Ah got it!